acquire/acquire.py

```diff
@@ -628,16 +628,42 @@ class Recents(Module):
     ]
 
 
+def recyclebin_filter(path: fsutil.TargetPath) -> bool:
+    return bool(path.stat().st_size >= (10 * 1024 * 1024))  # 10MB
+
+
 @register_module("--recyclebin")
+@module_arg(
+    "--large-files",
+    action="store_true",
+    help="Collect files larger than 10MB in the Recycle Bin",
+    default=False,
+)
+@module_arg(
+    "--no-data-files",
+    action="store_true",
+    help="Skip collection of data files in the Recycle Bin",
+    default=False,
+)
 class RecycleBin(Module):
-    DESC = "recycle bin metadata"
+    DESC = "recycle bin metadata and data files"
 
     @classmethod
     def _run(cls, target: Target, cli_args: argparse.Namespace, collector: Collector) -> None:
+        large_files_filter = None if cli_args.large_files else recyclebin_filter
+
+        if large_files_filter:
+            log.info("Skipping files in Recycle Bin that are larger than 10MB.")
+
         for fs, name, mountpoints in iter_ntfs_filesystems(target):
-            log.info("Acquiring recycle bin metadata from %s (%s)", fs, mountpoints)
+            log.info("Acquiring recycle bin from %s (%s)", fs, mountpoints)
+
+            patterns = ["$Recycle.bin/*/$I*", "Recycler/*/INFO2", "Recycled/INFO2"]
+
+            if not cli_args.no_data_files:
+                patterns.extend(["$Recycle.Bin/$R*", "$Recycle.Bin/*/$R*", "RECYCLE*/D*"])
 
-            patterns = ["$Recycle.bin/**/$I*", "Recycler/*/INFO2", "Recycled/INFO2"]
+            # FIXME: collect files using the large_files_filter
             for pattern in patterns:
                 for entry in fs.path().glob(pattern):
                     collector.collect_file(entry, outpath=fsutil.join(name, str(entry)))
```

Comment:
Probably to fix the `outpath` to be for the correct filesystem, which it looks like [677cbbb](https://github.com/fox-it/acquire/pull/86/commits/677cbbb60970745945eedbe3af32b86d40c2e630) may have now broken. Can you double check @Zawadidone? I believe this originally stored the output file for each individual filesystem as subdirectories in the resulting tar file of each filesystem. I'm not sure your current code still behaves that way.

Reply:
The function `collect_path` does not have the argument `outpath`. This function is used because the data files can also contain directories and previously only the info files were collected.

This is the output of the acquire container using `--recyclebin` using this branch, as shown the code does not behave the same.

```bash
target-fs -q MSEDGEWIN10_20230830090015.tar ls /
$Recycle.Bin
$Recycle.bin
fs0
fs1
```

This is the output of the acquire container `--recyclebin`  using the latest version of Acquire.

```bash
$ target-fs -q MSEDGEWIN10_20230830090306.tar ls /
$Recycle.bin
c:
sysvol

$ target-fs -q MSEDGEWIN10_20230830090306.tar ls /sysvol/\$recycle.bin/S-1-5-21-321011808-3761883066-353627080-1000
[...]
# Info files
```

Reply:
I have made an adjustment [`66d6d11` (#86)](https://github.com/fox-it/acquire/pull/86/commits/66d6d11bda36007452f1ea62fa28663169badc6c) to fix this, as shown below the code now behaves the same

```
target-fs -q MSEDGEWIN10_20230830091041.tar ls /
$Recycle.bin
c:
sysvol
```

Reply:
But it still shows a `$Recycle.bin` in the "root", isn't that still erroneous behaviour? May have been a bug in the original code though.

Reply:
That is a bug in the original code

Reply:
I will give it a try

Reply:
@Schamper or @Miauwkeru I couldn't find the the fix, can you take a look?

Reply:
This has fallen between the cracks a bit, sorry. I don't think I'll have time personally, but I raise this within the team.

Reply:
I tried to reproduce this on all the windows images I have locally, but couldn't. A code review also showed each fs should always get a `name` value which is used in the `outpath=` parameter. Also your latest code doesn't seem to do anything weird that may cause this.

Would it be possible for you to provide the image that shows this behaviour?

Reply:
I don't know what went wrong when executing the previous tests (https://github.com/fox-it/acquire/pull/86#discussion_r1309957195), but I couldn't reproduce the bug. The code works as intended.

```bash
# https://developer.microsoft.com/en-us/windows/downloads/virtual-machines/
acquire.py --recyclebin --registry --ntfs  MSEdge-Win10-VMware.vmwarevm/

target-fs MSEDGEWIN10_20231030172033.tar ls /
c:
sysvol

target-fs MSEDGEWIN10_20231030172033.tar ls /sysvol
$Boot
$Extend
$MFT
$Recycle.bin
$Secure:$SDS
documents and settings
users
windows
```